drivers/sensor/bh1750/bh1750.c

```diff
@@ -0,0 +1,140 @@
+{
+	struct bh1750_data *drv_data = dev->data;
+	const struct bh1750_config *cfg = dev->config;
+	uint8_t buf[2];
+	int rc = 0;
+
+	// In case of one time measurement we need to start a new measurement
+#ifdef CONFIG_BH1750_MODE_ONE_TIME
+	uint8_t buf_ctrl = BH1750_CTRL_MEAS;	
+	rc = i2c_write_dt(&cfg->i2c, &buf_ctrl, 1);
+	if(rc < 0){
+		LOG_ERR("Error writing in i2c bus: %d", rc);
+		return -ENXIO;
+	} else {
+		LOG_DBG("Starting a new measurement: %d", buf_ctrl);
+	}
+
+	// We need to wait for the new measurement to be taken
+	// According to datasheet page 5 section "Measurement mode explanation"
+	// 120 ms for H_RESOLUTION_MODE and H_RESOLUTION_MODE2
+	// 16 ms for L_RESOLUTION_MODE
+	#ifdef CONFIG_BH1750_H_RESOLUTION_MODE
+		k_msleep(120);
+	#elif CONFIG_BH1750_H_RESOLUTION_MODE2
+		k_msleep(120);
+	#elif CONFIG_BH1750_L_RESOLUTION_MODE
+		k_msleep(16);
+	#endif
+#endif
+
+	// read raw value data
+	rc = i2c_read_dt(&cfg->i2c, buf,2);
+
+	if (rc < 0) {
+		LOG_ERR("Failed to read data sample.");
+		return -EIO;
+	}
+
+	// We need to divide by 1.2 since it is the accuracy according to datasheet
+	float data_value = (((uint16_t)buf[0] << 8) | buf[1]) / 1.2;
+
+	// We need to divide by 2 in case of resolution mode 2
+	#ifdef CONFIG_BH1750_H_RESOLUTION_MODE2
+		data_value /= 2;
+	#endif
+
+	LOG_DBG("Light value: %f", data_value);
+	drv_data->ambient_light = data_value;
+
+	return 0;
+}
+
+static const struct sensor_driver_api bh1750_driver_api = {
+	.sample_fetch = bh1750_sample_fetch,
+	.channel_get = bh1750_channel_get,
+};
+
+
+
+int bh1750_init(const struct device *dev)
+{
+	const struct bh1750_config *cfg = dev->config;
+	int rc = 0;
+
+	if (!device_is_ready(cfg->i2c.bus)) {
+		LOG_ERR("Bus device is not ready");
+		return -ENODEV;
+	} else {
+		LOG_INF("Bus device is ready");
+	}
+
+	// Set mode and resolution
+	uint8_t buf = BH1750_CTRL_MEAS;	
+
+	rc = i2c_write_dt(&cfg->i2c, &buf, 1);
+	if(rc < 0){
+		LOG_ERR("Error writing in i2c bus: %d", rc);
+		return -ENXIO;
+	} else {
+		LOG_INF("The mode was set correctly: 0x%x", BH1750_CTRL_MEAS);
+	}
+
+	return 0;
+}
+
+
+#define BH1750_DEFINE(inst)									                        \
+	static struct bh1750_data bh1750_data_##inst;					                \
+												                                    \
+	/* pull instance configuration from Devicetree */                               \
+	static const struct bh1750_config bh1750_config_##inst = {			            \
+		.i2c = I2C_DT_SPEC_INST_GET(inst)						                    \
+	};						    				                                    \
+												                                    \
+    /* define a new device inst. with the data and config defined above*/           \
+	SENSOR_DEVICE_DT_INST_DEFINE(inst, bh1750_init, NULL,						    \
+			      &bh1750_data_##inst, &bh1750_config_##inst,			            \
+			      POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY,				            \
+			      &bh1750_driver_api);						                        \
```

Comment:
Please check your editor settings (tab is 8 spaces)

Reply:
Sorry, there were some spaces between tabs, I have checked it out and now it should be fine.